src/BugsnagUnity/FileManager.cs

```diff
@@ -106,30 +128,64 @@ internal static void CacheSession(string reportId)
             }
         }
 
+        internal static void CacheEvent(string reportId)
+        {
+            var eventReport = GetPendingEventReport(reportId);
+            if (eventReport == null)
+            {
+                return;
+            }
+            using (var stream = new MemoryStream())
+            using (var reader = new StreamReader(stream))
+            using (var writer = new StreamWriter(stream, new UTF8Encoding(false)) { AutoFlush = false })
+            {
+                SimpleJson.SerializeObject(eventReport.GetSerialisableEventReport(), writer);
+                writer.Flush();
+                stream.Position = 0;
+                var jsonString = reader.ReadToEnd();
+                var path = EventsDirectory + "/" + eventReport.Id + ".event";
+                WriteToDisk(jsonString, path);
+                CheckForMaxCachedEvents();
+            }
+        }
+
+        private static void CheckForMaxCachedEvents()
+        {
+            var filesCount = _cachedEvents.Length;
+            while (filesCount > _configuration.MaxPersistedEvents)
+            {
+                RemoveOldestFile(_cachedEvents);
+                filesCount = _cachedEvents.Length;
+            }
+        }
+
         private static void CheckForMaxCachedSessions()
         {
             var filesCount = _cachedSessions.Length;
             while(filesCount > _configuration.MaxPersistedSessions)
             {
-                RemoveOldestSession();
+                RemoveOldestFile(_cachedSessions);
                 filesCount = _cachedSessions.Length;
             }
         }
 
-        private static void RemoveOldestSession()
+        private static void RemoveOldestFile(string[] filePaths)
         {
-            var oldestSession = string.Empty;
+            var oldestFilePath = string.Empty;
             var oldestMillis = default(double);
```

Comment:
Minor and unrelated to this change but if you want 0 I think it's better to say 0!

Reply:
done 👍